File: tatami_test/lib/models/tache.dart
class Tache {
  final int? id;
  final String title;
  final String? description;
  final DateTime date;
  final int categoryId;
  final String etat;
  final int? ownerId;

  Tache({
    this.id,
    required this.title,
    this.description,
    required this.date,
    required this.categoryId,
    required this.etat,
    this.ownerId,
  });

  factory Tache.fromJson(Map<String, dynamic> json) => Tache(
    id: json['id'],
    title: json['title'] ?? '',
    description: json['description'],
    date: DateTime.parse(json['date']),
    categoryId: json['category_id'],
    etat: json['etat'] ?? 'en attente',
    ownerId:
        json['owner_id'] ??
        (json['owner'] != null ? json['owner']['id'] : null),
  );

  Map<String, dynamic> toJson() => {
    'title': title,
    'description': description,
    'date': date.toIso8601String().split('T')[0], // yyyy-mm-dd
    'category_id': categoryId,
    'etat': etat,
  };
}

File: laravel_flutter/app/Models/Tache.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Tache extends Model
{
    use HasFactory;

    protected $fillable = [
        'owner_id',
        'title',
        'description',
        'date',
        'category_id',
        'etat',
    ];

    protected $casts = [
        'date' => 'date',
    ];

    public function category()
    {
        return $this->belongsTo(Categorie::class, 'category_id');
    }

    public function owner()
    {
        return $this->belongsTo(User::class, 'owner_id');
    }

    public function sharedUsers()
    {
        return $this->belongsToMany(User::class, 'task_users', 'tache_id', 'user_id')->withTimestamps();
    }
}
